src/Aspire.Hosting/Health/ResourceHealthService.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Health;
+
+internal class ResourceHealthService(ILogger<ResourceHealthService> logger, HealthCheckService healthService) : IResourceHealthService
```

Comment:
We should be pushing heath status not pulling it. This should look at all resources and publish health updates via RNS